### PR TITLE
Remove Analyze

### PR DESCRIPTION
**Pull request by vertti**
## Summary

`Analyze` class is a remnant from pre-Strategy times, and now it doesn't contain really anything that should have a class of it's own: it's either stuff that should be in strategy or helpers that should not be a class (like parsing ticker data to dataframe for example).

This should also allow for much richer strategies as they can override the functions in the base strategy class.

This is work in progress so you guys know what's going on.

## Changelog

- [x] Move initialization of strategy to bot/backtest, out from Analyze
- [x] Remove pass-through functions in Analyze and call the strategy
- [x] Move functions that only use strategy to `IStrategy`
- [x] Move rest of the functions out from `Analyze` class
- [x] Remove `Analyze` class

### CODE REVIEW

**Comment by vertti** [comment]
Ok, so this progressing... still left to do:
- [ ] move few commented out tests from test_analyze to where they belong
- [ ] fix the strategy loading failing test in test_strategy.py

File: freqtrade/analyze.py [hunk 1]
```diff
@@ -2,270 +2,32 @@
...
-class SignalType(Enum):
-    """
-    Enum to distinguish between buy and sell signals
-    """
-    BUY = "buy"
-    SELL = "sell"
-
-
-class Analyze(object):
+def parse_ticker_dataframe(ticker: list) -> DataFrame:
```
Comment (xmatthias) [inline freqtrade/analyze.py]: what's the point in leaving a file for only one function? maybe we should find a "better" place for this ... or rename the file to "helper_functions_without_dependencies.py" ... (yes, i know I'm not good at naming things)

**Comment by xmatthias** [comment]
the parameters for ` _load_strategy` changed - extra_dir is now the 3rd directory ...

File: freqtrade/analyze.py (hunk 1 above)
Comment (vertti) [reply freqtrade/analyze.py]: Renamed `analyze.py` to `exchange_helpers.py` becauce it only parses exchange ticker data.

**Comment by vertti** [comment]
Scripts are now fixed. And analyze.py is renamed. This is ready for final review and merge.

File: freqtrade/analyze.py (hunk 1 above)
Comment (xmatthias) [reply freqtrade/analyze.py]: eventually we could move it to the exchange file too ... but that's a topic for another day :)

File: freqtrade/tests/conftest.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 from telegram import Chat, Message, Update
 
 from freqtrade import constants
-from freqtrade.analyze import Analyze
+from freqtrade.analyze import parse_ticker_dataframe
```
Comment (xmatthias) [inline freqtrade/tests/conftest.py]: if analyze.py is removed, this will fail

File: freqtrade/tests/conftest.py [hunk 3]
```diff
@@ -51,13 +51,13 @@ def get_patched_freqtradebot(mocker, config) -> FreqtradeBot:
     """
     # mocker.patch('freqtrade.fiat_convert.Market', {'price_usd': 12345.0})
     patch_coinmarketcap(mocker, {'price_usd': 12345.0})
-    mocker.patch('freqtrade.freqtradebot.Analyze', MagicMock())
+#    mocker.patch('freqtrade.freqtradebot.Analyze', MagicMock())
```
Comment (xmatthias) [inline freqtrade/tests/conftest.py]: i guess we should remove those if the class does not exist anymore...

File: freqtrade/tests/conftest.py (hunk 2 above)
Comment (vertti) [reply freqtrade/tests/conftest.py]: Fixed in 85fd4dd

File: freqtrade/tests/conftest.py (hunk 3 above)
Comment (vertti) [reply freqtrade/tests/conftest.py]: Fixed in 85fd4dd

**Pull request merged**